File: Compare Version Numbers.py
# https://leetcode.com/problems/compare-version-numbers/?envType=daily-question&envId=2024-05-03
# Compare Version Numbers

class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:

        v1 = list(map(int, version1.split('.')))
        v2 = list(map(int, version2.split('.')))
        
        while v1 and v1[-1] == 0:
            v1.pop()
        while v2 and v2[-1] == 0:
            v2.pop()
        
        max_len = max(len(v1), len(v2))
        v1.extend([0] * (max_len - len(v1)))
        v2.extend([0] * (max_len - len(v2)))
        
        for i in range(max_len):
            if v1[i] > v2[i]:
                return 1
            elif v1[i] < v2[i]:
                return -1
        
        return 0

File: Compare Version Numbers.js
// https://leetcode.com/problems/compare-version-numbers/?envType=daily-question&envId=2025-09-23
// Compare Version Numbers

/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    const v1 = version1.split('.').map(num => parseInt(num, 10));
    const v2 = version2.split('.').map(num => parseInt(num, 10));

    const n = Math.max(v1.length, v2.length);

    for (let i = 0; i < n; i++) {
        const rev1 = i < v1.length ? v1[i] : 0;
        const rev2 = i < v2.length ? v2[i] : 0;

        if (rev1 < rev2) return -1;
        if (rev1 > rev2) return 1;
    }
    return 0;
};